python/ray/air/dataset_iterator.py

```diff
@@ -0,0 +1,20 @@
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional, Union, Iterator
```

Comment:
Should `DatasetIterator` be defined in the `ray.data` subpackage?

Reply:
Hmm yeah originally I was thinking it should be AIR only, but actually I do think we want to push Datasets users towards this API for reading Datasets in the future.